proxy/src/control/destination/mod.rs

```diff
@@ -26,19 +30,35 @@ pub struct Resolver {
 #[derive(Debug)]
 struct ResolveRequest {
     authority: DnsNameAndPort,
+    respond: Respond,
+}
+
+#[derive(Debug)]
+struct Respond {
```

Comment:
Since in the mood to document everything, what exactly is this type for? I wonder if in describing its role, we can come up with a better name.

It's almost like an event handler. Is `OnUpdate` the worst thing I've thought of? Or `UpdateListener`?